plotutil/xyer.go

```diff
@@ -0,0 +1,70 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package plotutil
+
+import (
+	"fmt"
+
+	"github.com/gonum/plot/plotter"
+)
+
+// VecXY implements plotter.XYer.
+type VecXY struct {
+	X, Y []float64
+}
+
+// ZipXY is a convenience function to build a VecXY from slices with length checks.
+// Panics on length mismatch.
+func ZipXY(x, y []float64) plotter.XYer {
```

Comment:
I still don't understand the need for this function.

If eventually it does get included though, it should return z VecXY (or whatever it is called - renaming it `XYSlices` or `XYVecs` seems sensible since it would then be grouped with `XYer` and `XYs` in the godoc - same for the XYZ case).